File: Python/Sivitas_Akademik.py
# Import File
from Human import Human

# Membuat kelas Sivitas_Akademik dengan pewarisan dari Human


class Sivitas_Akademik(Human):

    # asal_universitas Sivitas_Akademik
    __asal_universitas = ""
    # email_edu Sivitas_Akademik
    __email_edu = ""

    # Constructor with default initialization
    def __init__(self, nama="", nik="", Jenis_Kelamin="", asal_univ="", email=""):
        # memanggil konstruktor dari kelas induk(superclass)
        super().__init__(nik, nama, Jenis_Kelamin)
        self.__asal_universitas = asal_univ
        self.__email_edu = email

    # Setter and Getter methods for private attributes
    # set asal Universitas
    def setUniv(self, asal_Univ):
        self.__asal_universitas = asal_Univ

    # Get asal Universitas
    def getUniv(self):
        return self.__asal_universitas

    # Set email
    def setEmail(self, email):
        self.__email_edu = email

    # Get email
    def getEmail(self):
        return self.__email_edu

File: C++/Sivitas_Akademik.cpp
#include "header.hh"

// Constructor
Sivitas_Akademik::Sivitas_Akademik() : Human()
{
    asal_universitas = "";
    email_edu = "";
}

// Constructor with base sivitas akademik attribute.
Sivitas_Akademik::Sivitas_Akademik(string nama, string nik, string gender) : Human(nama, nik, gender)
{
}

// Constructor with all attributes
Sivitas_Akademik::Sivitas_Akademik(string nama, string nik, string gender, string asal_universitas, string email_edu) : Human(nama, nik, gender)
{
    this->asal_universitas = asal_universitas;
    this->email_edu = email_edu;
}

// Getter and Setter
// Setter method for asal universitas
void Sivitas_Akademik::setUniv(string asal_universitas)
{
    this->asal_universitas = asal_universitas;
}

// Getter method for asal universitas
string Sivitas_Akademik::getUniv()
{
    return asal_universitas;
}

// Setter method for email
void Sivitas_Akademik::setEmail(string email_edu)
{
    this->email_edu = email_edu;
}

// Getter method for email
string Sivitas_Akademik::getEmail()
{
    return email_edu;
}

// Destructor
Sivitas_Akademik::~Sivitas_Akademik() {}
